### PR TITLE
Add ability to deep link directly to the upsell screen

### PR DESCRIPTION
**Pull request by flozia**
# References:

Jira: [MNTOR-3619](https://mozilla-hub.atlassian.net/browse/MNTOR-3619)

# Description

Add the ability to deep link directly to the upsell screen on the dashboard.

# Screenshot

image

# How to test

Visit `/user/dashboard/action-needed?dialog=subscriptions`.

# Checklist

- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added a unit test to test for potential regressions of this bug.
- [ ] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.

[MNTOR-3619]: https://mozilla-hub.atlassian.net/browse/MNTOR-3619?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/[[...slug]]/page.tsx [hunk 1]
```diff
@@ -51,6 +51,7 @@ type Props = {
   };
   searchParams: {
     nimbus_web_preview?: string;
+    dialog?: "subscriptions";
```
Comment (flozia) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/[[...slug]]/page.tsx:54]: The query param name and value are TBD. I’ve asked about the final convention [here](https://mozilla-hub.atlassian.net/browse/MNTOR-3619?focusedCommentId=938349).

**Review by Vinnl — APPROVED** [review]
Code looks good and works well. Possibly you could remove the query param from the URL when closing the dialog. (Or even derive the dialog-open state from the URL, i.e. adding the query parameter to open the dialog, but that's probably overkill.)

File: src/app/components/client/toolbar/Toolbar.tsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ export type Props = {
   lastScanDate: Date | null;
   experimentData: ExperimentData;
   children?: ReactNode;
+  showUpsellBadge?: boolean;
```
[the remark this answers is not in the document]
Comment (flozia) [reply src/app/components/client/toolbar/Toolbar.tsx]: That’s absolutely a better name for the prop — renamed in [c58b0c5](https://github.com/mozilla/blurts-server/pull/5096/commits/c58b0c50be7f21ceb2c93b18ef695d7a7dfbaa13).

[the remark this answers is not in the document]
Comment (flozia) [reply src/app/components/client/toolbar/Toolbar.tsx]: The `dialog` query param is now also being removed on closing the dialog with [5e3f299](https://github.com/mozilla/blurts-server/pull/5096/commits/5e3f29962140fd6a349a2239adbc0306e382865e).

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/Dashboard.test.tsx [hunk 3]
```diff
@@ -359,13 +362,29 @@ it("shows the premium upsell dialog of the Premium upsell badge open by default)
...
+    Meta,
+  );
+  render(<ComposedDashboard autoOpenUpsellDialog />);
+
+  expect(
+    screen.getByText("Turn on automatic data removal with ⁨Monitor Plus⁩"),
+  ).toBeInTheDocument();
+
+  const closeButtonIcon1 = screen.getByLabelText("Close modal");
+  await user.click(closeButtonIcon1.parentElement as HTMLElement);
```
Comment (Vinnl) [inline src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/Dashboard.test.tsx:385]: Don't we need an `expect` here to verify that it's actually closed, judging by the test description?

**Review by Vinnl — APPROVED** [review]
:rocket:

File: src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/Dashboard.test.tsx (hunk 3 above)
Comment (flozia) [reply src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/Dashboard.test.tsx:385]: Right right right: [3b30594](https://github.com/mozilla/blurts-server/pull/5096/commits/3b3059439796e8e9da11bb00203249d7074a7428).

**Pull request merged**